src/CSharpFunctionalExtensions.FluentAssertions/ResultTEExtensions.cs

```diff
@@ -0,0 +1,94 @@
+﻿using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+using FluentAssertions;
+
+namespace CSharpFunctionalExtensions.FluentAssertions;
+public static class ResultTEExtensions
+{
+    public static ResultTEAssertions<T, E> Should<T, E>(this Result<T, E> instance) => new(instance);
+}
+
+public class ResultTEAssertions<T, E> : ReferenceTypeAssertions<Result<T, E>, ResultTEAssertions<T, E>>
+{
+    public ResultTEAssertions(Result<T, E> instance) : base(instance) { }
+
+    protected override string Identifier => "Result{T}";
+
+    /// <summary>
+    /// Asserts a result is a success.
+    /// </summary>
+    /// <param name="because"></param>
+    /// <param name="becauseArgs"></param>
+    /// <returns></returns>
+    public AndConstraint<ResultTEAssertions<T, E>> Succeed(string because = "", params object[] becauseArgs)
+    {
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .Given(() => Subject)
+            .ForCondition(s => s.IsSuccess)
+            .FailWith("Expected Result to be successful but it failed");
+
+        return new AndConstraint<ResultTEAssertions<T, E>>(this);
+    }
+
+    /// <summary>
+    /// Asserts a result is a success with a specified value.
+    /// </summary>
+    /// <param name="value"></param>
+    /// <param name="because"></param>
+    /// <param name="becauseArgs"></param>
+    /// <returns></returns>
+    public AndConstraint<ResultTEAssertions<T, E>> SucceedWith(T value, string because = "", params object[] becauseArgs)
+    {
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .Given(() => Subject)
+            .ForCondition(s => s.IsSuccess)
+            .FailWith("Expected Result to be successful but it failed")
+            .Then
+            .Given(s => s.Value)
+            .ForCondition(v => v.Equals(value))
```

Comment:
Because of the earlier conditions that validated the `s.Value` is accessible here, you can silence the C# compiler by updating this to:

```csharp
.ForCondition(v => v!.Equals(value))
```

You can do the same for the `FailWith` method below